src/devices/Board/Board.cs

```diff
@@ -0,0 +1,407 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Pwm;
+using System.Device.Spi;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Iot.Device.Board
+{
+    public abstract class Board : MarshalByRefObject, IDisposable
+    {
+        private readonly PinNumberingScheme _defaultNumberingScheme;
+        private readonly object _pinReservationsLock;
+        private readonly Dictionary<int, List<PinReservation>> _pinReservations;
+        private bool _initialized;
+        private bool _disposed;
+
+        protected Board(PinNumberingScheme defaultNumberingScheme)
+        {
+            _defaultNumberingScheme = defaultNumberingScheme;
+            _pinReservations = new Dictionary<int, List<PinReservation>>();
+            _pinReservationsLock = new object();
+            _initialized = false;
+            _disposed = false;
+        }
+
+        ~Board()
+        {
+            Dispose(false);
+        }
+
+        public event Action<string, Exception> LogMessages;
+
+        protected bool Initialized
+        {
+            get
+            {
+                return _initialized;
+            }
+        }
```

Comment:
```suggestion
        protected bool Initialized => _initialized;
```

here and below

Reply:
I haven't gotten friend with this new syntax yet...